pkg/roll/run_update.go

```diff
@@ -244,19 +244,10 @@ func (u *update) rollLoop(podID types.PodID, hChecks <-chan map[types.NodeName]h
 					"nextRemove": nextRemove,
 					"nextAdd":    nextAdd,
 				}).Infof("Adding %d new nodes and removing %d old nodes", nextAdd, nextRemove)
-				if nextRemove > 0 {
-					err = u.rcs.CASDesiredReplicas(u.OldRC, oldNodes.Desired, oldNodes.Desired-nextRemove)
-					if err != nil {
-						u.logger.WithError(err).Errorln("Could not decrement old replica count")
-						break
-					}
-				}
-				if nextAdd > 0 {
-					err = u.rcs.CASDesiredReplicas(u.NewRC, newNodes.Desired, newNodes.Desired+nextAdd)
-					if err != nil {
-						u.logger.WithError(err).Errorln("Could not increment new replica count")
-						break
-					}
+				err = u.rcs.TransferReplicaCounts(u.NewRC, nextAdd, u.OldRC, nextRemove)
```

Comment:
interesting change - now zeroes are explicitly CASed to the same value instead of taking no action! I think it is an improvement.